src/Protocols.Abstractions/Features/IConnectionTransportFeature.cs

```diff
@@ -2,17 +2,24 @@
 using System.Collections.Generic;
 using System.IO.Pipelines;
 using System.Text;
+using System.Threading.Tasks;
 
 namespace Microsoft.AspNetCore.Protocols.Features
 {
     public interface IConnectionTransportFeature
     {
         PipeFactory PipeFactory { get; }
 
-        IPipeConnection Connection { get; set; }
+        IPipeConnection Transport { get; set; }
+
+        IPipeConnection Application { get; set; }
 
         IScheduler InputWriterScheduler { get; }
 
         IScheduler OutputReaderScheduler { get; }
+
+        Task ConnectionAborted { get; }
+
+        Task ConnectionClosed { get; }
```

Comment:
Why can't a single Task represent aborted and closed?

Didn't we intend to communicate when/how the connection was closed via the IPipeConnections?